pkg/kubelet/cm/memorymanager/state/state_checkpoint_test.go

```diff
@@ -214,63 +157,137 @@ func TestCheckpointStateRestore(t *testing.T) {
 }
 
 func TestCheckpointStateStore(t *testing.T) {
+	expectedState := &stateMemory{
+		assignments: ContainerMemoryAssignments{
+			"pod": map[string][]Block{
+				"container1": []Block{
+					{
+						NUMAAffinity: 0,
+						Type:         core.ResourceMemory,
+						Size:         1024,
+					},
+				},
+			},
+		},
+		machineState: MemoryMap{
+			0: map[core.ResourceName]MemoryTable{
+				core.ResourceMemory: MemoryTable{
+					Allocatable:    1536,
+					Free:           1024,
+					Reserved:       512,
```

Comment:
As I understand correctly, if `Block` is set, in this case to 1024, then MemoryMap should be updated, in this case to 1024. Other values should also be updated.

Reply:
done